app/src/main/java/org/phenoapps/verify/UriHandler.java

```diff
@@ -0,0 +1,217 @@
+package org.phenoapps.verify;
+
+import android.content.ContentUris;
+import android.content.Context;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Environment;
+import android.provider.DocumentsContract;
+import android.provider.OpenableColumns;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class UriHandler {
```

Comment:
As we discussed, this should be a static utility class so there is only ever one instance.

Reply:
Also add a descriptive class-level comment that describes the usages and references.